### PR TITLE
feature(downloads): adds historical us and state csv downloads

### PR DESCRIPTION
**Pull request by scandycuz**
Adds feature for issue #1033 (Surface downloadable CSVs)

Please let me know if anything should be done differently. One thing I still had some questions about is updating the GraphQL content, such as the preamble on the data page. I imagine it's in a database somewhere? 

I was also unsure how to add the '← Our Data' navigation button above the page title, although I was thinking it relied on some GraphQL data as well.

### CODE REVIEW

**Comment by kevee** [comment]
Thanks, @scandycuz! Here's my thoughts:

The preamble is actually stored in Netlify, so if you add this component:

```
<ContentfulContent
      content={
        data.contentfulSnippet.childContentfulSnippetContentTextNode
          .childMarkdownRemark.html
      }
      id={data.contentfulSnippet.contentful_id}
    />```

Then this query to the page:

```
contentfulSnippet(slug: { eq: "download-page-preamble" }) {
      contentful_id
      childContentfulSnippetContentTextNode {
        childMarkdownRemark {
          html
        }
      }
    }
```

You should be good to go!

For the "return link," set the `navigation` prop on the `Layout` to: `{data.contentfulNavigationGroup.pages}`

Then add this query to the page to snag the "data" navigation group:

```
contentfulNavigationGroup(slug: { eq: "data" }) {
      pages {
        ... on ContentfulPage {
          title
          link: slug
        }
        ... on ContentfulNavigationLink {
          title
          link: url
        }
      }
    }
```

**Comment by scandycuz** [comment]
Thanks @kevee! I've made the changes. 

It does seem that the 'return link' still doesn't display, but I think it may just need to be added to the GraphQL data?

**Comment by amandafrench** [comment]
Thanks so much for doing this much-needed work, @scandycuz!

**Comment by scandycuz** [comment]
@amandafrench of course, no problem!

**Comment by kevee** [comment]
Suggestions from @amandafrench:

- [ ] "US History" --> "US Data" (or possibly "US COVID-19 Data")
- [ ] "Download all history data" --> "Download all available US data"
- [ ] "State History" --> "State Data" (or possibly "State COVID-19 Data")
- [ ] "Download all states" --> "Download all available data for all states"
- [ ] "Download single state" --> "Download all available data for a single state"

**Comment by scandycuz** [comment]
@kevee Happy to make the changes. Also, currently the headers are hard coded, but should I link them to Contentful instead? If so, I can add the content there, or just link to it if it would be easier for someone else to create it.

**Comment by kevee** [comment]
Let's make them hard-coded, the contentful part should just be the top part of the page.

**Comment by scandycuz** [comment]
Sounds good, thanks

File: src/pages/data/download.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+      id={data.contentfulSnippet.contentful_id}
+    />
+
+    <h2>US Data</h2>
+    <DownloadLink
+      desc="Download all available US data"
+      path="/api/v1/us/daily.csv"
+    />
+    <Panel label="Field Descriptions">
+      <Fields schema="Us" />
```
Comment (scandycuz) [inline src/pages/data/download.js:29]: @kevee Is it possible that the name of the schema could change? If so I could get the schema name from the path, e.g.
```
import definition from '../../../_api/v1/openapi.json'
...
  <Fields
    schema={
      definition.paths['/api/v1/states/daily.{format}'].get.responses[200].content['text/csv']
        .schema.items.$ref.split('/').pop()
    }
  />
...
```

**Pull request merged**